vaadin-menu-bar-flow/src/main/resources/META-INF/resources/frontend/menubarConnector.js

```diff
@@ -13,42 +13,49 @@
-      return;
-    }
-    menubar.$connector = {
-
-      updateButtons: function () {
-        if (!menubar.shadowRoot) {
-          // workaround for https://github.com/vaadin/flow/issues/5722
-          setTimeout(() => menubar.$connector.updateButtons());
-          return;
-        }
 
-        // Propagate disabled state from items to parent buttons
-        menubar.items.forEach(item => item.disabled = item.component.disabled);
+(function () {
+  const tryCatchWrapper = function (callback) {
+    return window.Vaadin.Flow.tryCatchWrapper(callback, 'Vaadin Menu Bar', 'vaadin-menu-bar-flow');
+  };
 
-        // Remove hidden items entirely from the array. Just hiding them
-        // could cause the overflow button to be rendered without items.
-        // resetContent needs to be called to make buttons visible again.
-        //
-        // The items-prop needs to be set even when all items are visible
-        // to update the disabled state and re-render buttons.
-        menubar.items = menubar.items.filter(item => !item.component.hidden);
+  window.Vaadin.Flow.menubarConnector = {
+    initLazy: function (menubar) {
+      if (menubar.$connector) {
+        return;
+      }
+      menubar.$connector = {
 
-        // Propagate click events from the menu buttons to the item components
-        menubar._buttons.forEach(button => {
-          if (button.item && button.item.component) {
-            button.addEventListener('click', e => {
-              if (e.composedPath().indexOf(button.item.component) === -1) {
-                button.item.component.click();
-              }
-            });
+        updateButtons: tryCatchWrapper(function () {
+          if (!menubar.shadowRoot) {
+            // workaround for https://github.com/vaadin/flow/issues/5722
+            setTimeout(() => menubar.$connector.updateButtons());
+            return;
           }
-        });
+
+          // Propagate disabled state from items to parent buttons
+          menubar.items.forEach(item => item.disabled = item.component.disabled);
+
+          // Remove hidden items entirely from the array. Just hiding them
+          // could cause the overflow button to be rendered without items.
+          // resetContent needs to be called to make buttons visible again.
+          //
+          // The items-prop needs to be set even when all items are visible
+          // to update the disabled state and re-render buttons.
+          menubar.items = menubar.items.filter(item => !item.component.hidden);
+
+          // Propagate click events from the menu buttons to the item components
+          menubar._buttons.forEach(button => {
+            if (button.item && button.item.component) {
+              button.addEventListener('click', e => {
+                if (e.composedPath().indexOf(button.item.component) === -1) {
+                  button.item.component.click();
+                }
+              });
+            }
+          });
+        })
       }
```

Comment:
This line should end in `;`. This is the end of `menubar.$connector = {` assignment.

Reply:
Updated.